programs/fileio.c

```diff
@@ -1999,16 +2007,105 @@ int FIO_compressMultipleFilenames(FIO_ctx_t* const fCtx,
 /* **************************************************************************
  *  Decompression
  ***************************************************************************/
+#define DECOMPRESSION_MAX_WRITE_JOBS    (10)
+
+typedef struct {
+    POOL_ctx* writerPool;
+    ZSTD_pthread_mutex_t writeJobsMutex;
+    void* jobs[DECOMPRESSION_MAX_WRITE_JOBS];
+    volatile int availableWriteJobs;
+    int totalWriteJobs;
+    FILE* dstFile;
+    unsigned storedSkips;
+    FIO_prefs_t* prefs;
+} write_pool_ctx_t;
+
+typedef struct {
+    write_pool_ctx_t *ctx;
+    FILE* dstFile;
+    void *buffer;
+    size_t bufferSize;
+    size_t usedBufferSize;
+} write_job_t;
+
 typedef struct {
     void*  srcBuffer;
     size_t srcBufferSize;
     size_t srcBufferLoaded;
-    void*  dstBuffer;
-    size_t dstBufferSize;
     ZSTD_DStream* dctx;
-    FILE*  dstFile;
+    write_pool_ctx_t *writePoolCtx;
 } dRess_t;
 
+static write_job_t *FIO_createWriteJob(write_pool_ctx_t *ctx) {
+    void *buffer;
+    write_job_t *job;
+    job = (write_job_t*) malloc(sizeof(write_job_t));
+    buffer = malloc(ZSTD_DStreamOutSize());
+    if(!job || !buffer)
+        EXM_THROW(101, "Allocation error : not enough memory");
+    job->buffer = buffer;
+    job->bufferSize = ZSTD_DStreamOutSize();
+    job->usedBufferSize = 0;
+    job->dstFile = NULL;
+    job->ctx = ctx;
+    return job;
+}
+
+static void FIO_writePoolCreateThreadPool(write_pool_ctx_t *ctx, const FIO_prefs_t *prefs) {
```

Comment:
nit: naming: Can you name the functions `FIO_WritePool_createThreadPool()`? Or just `WritePool_createThreadPool()` since they're all `static`. That makes it clear its the function `createThreadPool()` for the object/concept/group `WritePool`.

Reply:
Fixed